src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,554 @@
+    """From simple (subject, predicate, object) edge tuples, create dictionaries with mondo mappings and labels
+
+    Populates: subject_source_id, subject_mondo_id, subject_mondo_label, object_source_id, object_mondo_id,
+    object_mondo_label """
+    edges2 = []
+    for e in edges:
+        d = {}
+        terms = {'subject': e[0], 'object': e[2]}
+        for position, _id in terms.items():
+            d_i = {f'{position}_source_id': _id, f'{position}_mondo_id': None, f'{position}_mondo_label': None}
+            try:
+                mapping: Dict = source_data_map[_id]
+                for field in ['mondo_id', 'mondo_label']:
+                    d_i[f'{position}_{field}'] = mapping[field]
+            except KeyError:
+                pass  # the term only exists in source and is not in Mondo yet, so has no Mondo ID/label
+            d = d | d_i
+        edges2.append(d)
+    return edges2
+
+
+# todo: temp: https://github.com/monarch-initiative/mondo-ingest/issues/92
+def _report_duplicate_exact_mappings_by_direction(df: pd.DataFrame, by=['mondo', 'source'][0]):
+    """Report duplciate exact mappings in mondo.sssom.tsv, based on 1 mondo ID to 2+ source terms, or vice versa"""
+    if by not in ['mondo', 'source']:
+        raise ValueError("'by' must be either of 'mondo' or 'source'.")
+    field = 'subject_id' if by == 'mondo' else 'object_id'
+    filename_prefix = 'mondo-source' if by == 'mondo' else 'source-mondo'
+    i = 1
+    dups = []
+    for prefix, df_i in df.groupby('obj_prefix'):
+        j = 1
+        # Collect all mappings by ID
+        row_map_by_mondo_id = {}
+        for _index, row in df_i.iterrows():
+            row = dict(row)
+            if row[field] not in row_map_by_mondo_id:
+                row_map_by_mondo_id[row[field]] = []
+            row_map_by_mondo_id[row[field]].append(row)
+        # Filter and name duplicates
+        for mappings in row_map_by_mondo_id.values():
+            if len(mappings) > 1:
+                dups.extend([d | {'dupe_id': i, 'source_dupe_id': f'{prefix}_{j}'} for d in mappings])
+                i += 1
+                j += 1
+    out_df = pd.DataFrame(dups).rename(columns={'obj_prefix': 'source'})[[
+        'dupe_id', 'source', 'source_dupe_id', 'subject_id', 'object_id', 'predicate_id', 'subject_label']].sort_values(
+        ['dupe_id', 'source', 'source_dupe_id', 'object_id'])
+    out_df.to_csv(TMP_DIR / f'{filename_prefix}-dups.sssom.tsv', sep='\t', index=False)
+
+
+# todo: temp: https://github.com/monarch-initiative/mondo-ingest/issues/92
+def _report_duplicate_exact_mappings(df: pd.DataFrame):
+    """Report duplciate exact mappings in mondo.sssom.tsv"""
+    _report_duplicate_exact_mappings_by_direction(df, 'mondo')
+    _report_duplicate_exact_mappings_by_direction(df, 'source')
+
+
+def _check_inputs_in_sync(mondo_db: SqlImplementation, ss_df: pd.DataFrame):
```

Comment:
## `mondo.owl` and `mondo.sssom.tsv` out of sync (@matentzn review)
@matentzn Review implementation and questions here, as well as issue #384.

<details><summary>Details</summary>
<p>

As written in the issue:
> I found 971 terms that were in mondo.db (from mondo.owl) that weren't in mondo.sssom.tsv.
> I found 2,024 terms that were in mondo.sssom.tsv but not in mondo.db.

The implementation for this changed. Right now I'm just doing a static check of all of the Mondo IDs in `mondo.db` and `mondo.sssom.tsv`.

The implementation I had before gave more information based on which terms were missing by source, but I didn't think that information was important, so I changed it. 

Before, I was printing a warning for sources where the number of out of sync terms were greater than 200. Now I'm saving a report and running the same check every time (source agnostic), but if it's greater than 200 I'm still warning. But now I'm also always saving this report: [out_of_sync_mondo_ids.tsv.zip](https://github.com/monarch-initiative/mondo-ingest/files/13520221/out_of_sync_mondo_ids.tsv.zip)

**Questions**
1. @matentzn If this is a static QC check that does not depend on any source nor on this synchronization work, should I make a separate PR for this? Should I add this to some kind of QC command or a test (maybe eventually QC checks will be run as Python tests?)?
2. @matentzn Do you want me to change this back to how I had it before? Some kind of breakdown by source, rather than simply checking `mondo.db` and `mondo.sssom.tsv` against each other? Or should I do both--both checking those against each other, and some kind of breakdown by source?

</p>
</details>

Reply:
For reports that require discussion, its better to have them on Google sheets so I can click immediately;

> I found 971 terms that were in mondo.db (from mondo.owl) that weren't in mondo.sssom.tsv.

For now, its best to look at once example and then decide if it is an issue. I would have expected around 1000K terms in Mondo that do not have any associated mappings, and therefore do not show up in the mondo.sssom.tsv. So this one seems ok to me.

> I found 2,024 terms that were in mondo.sssom.tsv but not in mondo.db.

Again it would be good to look at some examples. Out-of-syncness is only 1 hypothesis; another one is that the sssom file contains 2000 obsolete classes that are not present in mondo.db (which is interesting to know, as I would not agree with the oak design in this case). 


I fully agree with a simple QC check that ensures a minimum of X classes are being processed, ideally on a per resource basis; but for now I suspect the cases you found have good explanations.

Reply:
Thanks for response! You've approved a QC check so I think I'm going to move this work out of this PR. In that case, let's continue the discussion in the issue #384.

Reply:
We have been discussing in issue but now need to figure out what to do with warnings / QC check.

Reply:
Moved into #385